CLUtils/CogSLanguageUtilities.Tests/TestUtils/DocumentTreeComparer.cs

```diff
@@ -0,0 +1,61 @@
+﻿using Microsoft.CogSLanguageUtilities.Definitions.Models.Document;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.CogSLanguageUtilities.Tests.TestUtils
+{
+    public class DocumentTreeComparer : IEqualityComparer<DocumentTree>
+    {
+        public bool Equals(DocumentTree x, DocumentTree y)
+        {
+            if(x == null && y == null)
+            {
+                return true;
+            }
+            else if (x == null || y == null)
+            {
+                return false;
+            }
+            var list = x.RootSegment.Children.Zip(y.RootSegment.Children, (e1, e2) => new { e1, e2 });
```

Comment:
Why not loop with an index instead of doing a `Zip` then `foreach`? You also need to compare the number of children.

Reply:
refactored